python/pythonlab/neighborhood/neighborhood/painter.py

```diff
@@ -46,43 +61,65 @@ def paint(self, color):
     Args:
       color (str): The color to paint the square.
     """
-    signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, NeighborhoodSignalKey.PAINT.value, {'color': color})
-    print(signal_message.get_formatted_message())
+    self._send_signal(NeighborhoodSignalKey.PAINT.value, {'id': self.id, 'color': color})
 
   def scrape_paint(self):
     """
     Removes all the paint off the square the painter is on.
     """
-    signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, NeighborhoodSignalKey.REMOVE_PAINT.value)
-    print(signal_message.get_formatted_message())
+    self._send_signal(NeighborhoodSignalKey.REMOVE_PAINT.value, {'id': self.id})
 
-  def get_my_paint(self):
-    """
-    Returns the amount of paint the painter has.
-    """
-    return self.remaining_paint
-  
   def hide_painter(self):
     """
     Hides the painter on the screen.
     """
-    signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, NeighborhoodSignalKey.HIDE_PAINTER.value)
-    print(signal_message.get_formatted_message())
+    self._send_signal(NeighborhoodSignalKey.HIDE_PAINTER.value, {'id': self.id})
 
   def show_painter(self):
     """
     Shows the painter on the screen.
     """
-    signal_message = NeighborhoodSignalMessage(SignalMessageType.PAINTER, NeighborhoodSignalKey.SHOW_PAINTER.value)
-    print(signal_message.get_formatted_message())
+    self._send_signal(NeighborhoodSignalKey.SHOW_PAINTER.value, {'id': self.id})
 
   def take_paint(self):
     """
-    The Painter adds a single unit of paint to their personal bucket. The counter on the bucket on
-    the screen goes down. If the painter is not standing on a paint bucket, nothing happens.
+    The Painter adds a single unit of paint to their personal bucket.
+    The counter on the bucket on the screen goes down.
+    If the painter is not standing on a paint bucket, nothing happens.
+    """
+    self._send_signal(NeighborhoodSignalKey.TAKE_PAINT.value, {'id': self.id})
+
+  def show_buckets(self):
+    """
+    Show all the paint buckets on the screen.
+    """
+    self._send_signal(NeighborhoodSignalKey.SHOW_BUCKETS.value)
+
+  def hide_buckets(self):
+    """
+    Hide all the paint buckets on the screen.
+    """
+    self._send_signal(NeighborhoodSignalKey.HIDE_BUCKETS.value)
+
+  def send_initialization_message(self, x, y, direction, paint):
+    print(self._get_initialization_message(x, y, direction, paint).get_formatted_message())
+
+  def _get_initialization_message(self, x, y, direction, paint):
```

Comment:
Passing values as params because when I referred to `self.direction`, it was not setting properly since `Direction`'s instance was not yet available.